tiledb/sm/config/config.cc

```diff
@@ -128,7 +128,7 @@ const std::string Config::VFS_S3_AWS_SECRET_ACCESS_KEY = "";
 const std::string Config::VFS_S3_AWS_SESSION_TOKEN = "";
 const std::string Config::VFS_S3_AWS_ROLE_ARN = "";
 const std::string Config::VFS_S3_AWS_EXTERNAL_ID = "";
-const std::string Config::VFS_S3_AWS_LOAD_FREQUENCY = "";
+const std::string Config::VFS_S3_AWS_LOAD_FREQUENCY = "3600";
```

Comment:
Does this need to be set to the frequency the role is allowed to take? The role has a configured maximum time its allowed to be assumed before refresh, what if it is less than 1 hour? Does this setting of 1 hour break it?

Reply:
If we set the load_frequency(duration_seconds) higher than the configured maximum time limit, the operation of  AssumeRoleRequest will be failed. I will change it back to empty

Reply:
As a followup, if its empty what does this mean for the refresh? Does empty mean the SDK defaults to the role's configured max, or does empty mean disabled?

Reply:
If it is empty, our codes set the load_frequency to be Aws::Auth;:DEFAULT_CREDS_LOAD_FREQ_SECONDS which is 900 seconds. This is the minimum value we can set as load_frequency. If we set a value less than 900 or larger than 43200(12 hours), the AssumeRoleRequest will fail. Also if we set a value larger than the maximum value set by cloud administrator, it will also fail. When the load_frequency is 900 seconds, it will refresh every 14 minutes(900-60). AWS sdk has a constant ACCOUNT_FOR_LATENCY for 60.

Reply:
@bdeng-xt thanks for the explanation. So the end result is now a token refreshes every 14 minutes unless overridden by the user? So this should cover more or less all scenarios right because the fresh happens before expiration in all cases? Just want to make sure I am understanding correctly.

Reply:
Yes, 900 seconds as load_frequency should works for all scenarios. If we know the time limit set by cloud administrator, we could override it with a larger value. Last week I did some very simple performance test of AssumeRole in LoadCredentialsFromSTS, it took less than 10 milliseconds. so if the time limit is set to one hour and we refresh every 14 minutes, we are going to have three more extra refresh temporary credentials.  This should not be a big performance issue. Also I checked related codes in the aws-sdk code, lock is used when refreshing credentials, so thread should not have safety issues. I didn't use a very big data set to test it .  I only used some very simple code from our incomplete query code and thread.sleeping to simulate scenarios.

Reply:
Thanks for the detailed explanation, this will help when I explain to others.